modules/light-clients/08-wasm/types/client_state.go

```diff
@@ -249,20 +256,40 @@ func call[T ContractResult](ctx sdk.Context, clientStore sdk.KVStore, cs *Client
 	var output T
 	encodedData, err := json.Marshal(payload)
 	if err != nil {
-		return output, errorsmod.Wrapf(err, "failed to marshal wasm contract payload")
+		return output, errorsmod.Wrapf(err, "failed to marshal payload for wasm execution")
 	}
 	out, err := callContract(ctx, clientStore, cs.CodeHash, encodedData)
 	if err != nil {
 		return output, errorsmod.Wrapf(err, "call to wasm contract failed")
 	}
 	if err := json.Unmarshal(out.Data, &output); err != nil {
-		return output, errorsmod.Wrapf(err, "failed unmarshal wasm contract payload")
+		return output, errorsmod.Wrapf(err, "failed to unmarshal result of wasm execution")
 	}
 	if !output.Validate() {
-		return output, errorsmod.Wrapf(errors.New(output.Error()), "error occurred while calling contract with code hash %s", hex.EncodeToString(cs.CodeHash))
+		return output, errorsmod.Wrapf(errors.New(output.Error()), "error occurred while executing contract with code hash %s", hex.EncodeToString(cs.CodeHash))
 	}
 	if len(out.Messages) > 0 {
 		return output, errorsmod.Wrapf(ErrWasmSubMessagesNotAllowed, "code hash (%s)", hex.EncodeToString(cs.CodeHash))
 	}
 	return output, nil
 }
+
+// wasmQuery queries the contract with the given payload and writes the result to output.
+func wasmQuery[T ContractResult](ctx sdk.Context, clientStore sdk.KVStore, cs *ClientState, payload any) (T, error) {
+	var output T
+	encodedData, err := json.Marshal(payload)
+	if err != nil {
+		return output, errorsmod.Wrapf(err, "failed to marshal payload for wasm query")
+	}
+	out, err := queryContract(ctx, clientStore, cs.CodeHash, encodedData)
+	if err != nil {
+		return output, errorsmod.Wrapf(err, "call to wasm contract failed")
```

Comment:
to distinguish between the two:

```suggestion
		return output, errorsmod.Wrapf(err, "query to wasm contract failed")
```